fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java

```diff
@@ -868,6 +868,22 @@ public PlanFragment visitPhysicalValues(OptExpression optExpr, ExecPlan context)
             }
         }
 
+        // return true if all leaf offspring of the root are OlapScanNodes.
+        public static boolean allLeafAreOlapScanNodes(PlanNode root) {
+            if (root instanceof OlapScanNode) {
+                return true;
+            }
+            if (root.getChildren().isEmpty()) {
```

Comment:
Needn't check child is empty

Reply:
a root has no children means it is a leaf node, according to above if stmt, the this leaf node is not a OlapScanNode, so the fragment has leaf nodes that is not OlapScanNode.